modules/imap/functions.php

```diff
@@ -857,19 +857,31 @@ function imap_refresh_oauth2_token($server, $config) {
  * @return int count of messages moved
  */
 if (!hm_exists('imap_move_same_server')) {
-function imap_move_same_server($ids, $action, $hm_cache, $dest_path) {
+function imap_move_same_server($ids, $action, $hm_cache, $dest_path, $screen_emails=false) {
     $moved = array();
     $keys = array_keys($ids);
     $server_id = array_pop($keys);
     $cache = Hm_IMAP_List::get_cache($hm_cache, $server_id);
     $imap = Hm_IMAP_List::connect($server_id, $cache);
     foreach ($ids[$server_id] as $folder => $msgs) {
         if (imap_authed($imap) && $imap->select_mailbox(hex2bin($folder))) {
-            if ($imap->message_action(mb_strtoupper($action), $msgs, hex2bin($dest_path[2]))) {
+            if ($screen_emails) {
                 foreach ($msgs as $msg) {
                     $moved[]  = sprintf('imap_%s_%s_%s', $server_id, $msg, $folder);
+                    $email = current(array_column(process_address_fld($imap->get_message_headers($msg)['From']), "email"));
+                    $uid = $imap->search('ALL', false, array(array('FROM', $email)));
+                    if ($imap->message_action(mb_strtoupper($action), $uid, hex2bin($dest_path[2]))) {
```

Comment:
What is the purpose of this block? Move all emails from this sender to the destination folder? Is this the intended behavior?

Reply:
Targeted behaviour? No, but we found it interesting to do so for an address that we find not interesting.

Reply:
Then, is user informed that he would actually move many messages at once? Shouldn't this be a configuration or user choice somewhere in the UI?

Reply:
> Then, is user informed that he would actually move many messages at once? Shouldn't this be a configuration or user choice somewhere in the UI?
I find this suggestion very interesting. I think that's how I'm going to do it.

Reply:
![move-message-in-screen-email](https://github.com/user-attachments/assets/4599bf7a-8fdd-48e5-96f6-3223702595d6)